### ISSUE TITLE
[P3] Streaming STT with word-level timestamps

### ISSUE THREAD

**Issue by drizki** [drizki/voxclaw#109]
## Goal
Add streaming STT support with real-time partial results and word-level timestamps.

## Current Behavior
STT waits for complete utterance (after VAD speech-end) before transcribing.

## Proposed Behavior
Stream partial transcriptions as speech is detected, with word timestamps.

### 1. Extend SttResult
```rust
pub struct SttResult {
    pub text: String,
    pub words: Vec<WordInfo>,
    pub is_final: bool,  // NEW: false for partial results
    pub confidence: Option<f32>,
}

pub struct WordInfo {
    pub word: String,
    pub start_ms: u32,
    pub end_ms: u32,
    pub confidence: Option<f32>,
}
```

### 2. Extend Stt Trait
```rust
#[async_trait]
pub trait Stt {
    async fn transcribe(&mut self, samples: &[f32]) -> Result<SttResult>;
    fn sample_rate(&self) -> u32;
    
    // NEW: Streaming API
    async fn transcribe_streaming(
        &mut self,
        samples: &[f32],
    ) -> Result<impl Stream<Item = SttResult>>;
}
```

### 3. WhisperStt Updates
- Use Whisper's timestamps feature
- Emit partial results during long utterances
- Configure chunk size for latency/accuracy tradeoff

### 4. State Machine Updates
- New event: `PartialTranscript(String)`
- Allow brain to see partial transcripts (for faster response)

### 5. Protocol Updates
Add to WebSocket protocol:
```json
{"type": "partial_transcript", "text": "Hello...", "is_final": false}
{"type": "transcript", "text": "Hello world", "is_final": true, "words": [...]}
```

## Benefits
- Lower perceived latency
- Real-time captions/subtitles
- Interrupt detection (user starts speaking while TTS playing)
- Word-level highlighting in UI

## Acceptance Criteria
- [ ] Partial transcripts emitted during speech
- [ ] Word timestamps in final result
- [ ] Configurable chunk size
- [ ] Backwards compatible (non-streaming still works)
- [ ] Tests for streaming behavior

## References
- Whisper word timestamps: `--word_timestamps` flag
- Current: `src/stt/whisper/mod.rs`
